WhateverGreen/kern_igfx.hpp

```diff
@@ -499,6 +499,95 @@ class IGFX {
 		static bool forceWakeWaitAckFallback(uint32_t, uint32_t, uint32_t);
 		static void forceWake(void*, uint8_t set, uint32_t dom, uint32_t);
 	} ForceWakeWorkaround;
+	
+	/**
+	 *  Interface of a submodule to fix Intel graphics drivers
+	 */
+	class PatchSubmodule {
+	public:
+		/**
+		 *  Virtual destructor
+		 */
+		virtual ~PatchSubmodule() = default;
+		
+		/**
+		 *  True if this submodule should be enabled
+		 */
+		bool enabled {false};
+		
+		/**
+		 *  True if this submodule requires patching the framebuffer driver
+		 */
+		bool requiresPatchingFramebuffer {false};
+		
+		/**
+		 *  True if this submodule requires patching the graphics acceleration driver
+		 */
+		bool requiresPatchingGraphics {false};
+		
+		/**
+		 *  Initialize any data structure required by this submodule if necessary
+		 */
+		virtual void init() = 0;
+		
+		/**
+		 *  Release any resources obtained by this submodule if necessary
+		 */
+		virtual void deinit() = 0;
+		
+		/**
+		 *  Setup the fix and retrieve the device information if necessary
+		 *
+		 *  @param patcher  KernelPatcher instance
+		 *  @param info     Information about the graphics device
+		 *  @note This function is called when the main IGFX module processes the kernel.
+		 */
+		virtual void processKernel(KernelPatcher &patcher, DeviceInfo *info) = 0;
+
+		/**
+		 *  Process the framebuffer kext, retrieve and/or route functions if necessary
+		 *
+		 *  @param patcher KernelPatcher instance
+		 *  @param index   kinfo handle
+		 *  @param address kinfo load address
+		 *  @param size    kinfo memory size
+		 *  @note This funbction is called when the main IGFX module processes the kext.
+		 */
+		virtual void processFramebufferKext(KernelPatcher &patcher, size_t index, mach_vm_address_t address, size_t size) = 0;
+		
+		/**
+		 *  Process the accelerator kext, retrieve and/or route functions if necessary
+		 *
+		 *  @param patcher KernelPatcher instance
+		 *  @param index   kinfo handle
+		 *  @param address kinfo load address
+		 *  @param size    kinfo memory size
+		 *  @note This funbction is called when the main IGFX module processes the kext.
+		 */
+		virtual void processAcceleratorKext(KernelPatcher &patcher, size_t index, mach_vm_address_t address, size_t size) = 0;
```

Comment:
Is Accelerator kext — graphics kext? This is a bit not obvious as everything else uses graphics term. Maybe rename?

Reply:
Yes, it is the graphics kext. Sure, to make things consistent. 
Fixed.